lib/CreepPlayers/Behaviors/MoveToExitHallwayDumb.js

```diff
@@ -0,0 +1,39 @@
+
+var MoveUtils = require('../../Utils/MoveUtils');
+var Base = require('./Base');
+var Direction = require('../../GameObjects/Direction');
+
+/**
+ *
+ * @param creepController @type CreepController
+ * @constructor
+ */
+function MoveToExitDumb(creepController) {
+    this.cc = creepController;
+}
+
+MoveToExitDumb.prototype = Object.create(Base.prototype);
+
+MoveToExitDumb.prototype.act = function() {
+    var toExit = this.cc.senseDirectionToExit();
+    var toPlayer = this.cc.getDirectionToPlayer();
+
+    var blocking = false;
+    if (!this.cc.canMove(Direction.NORTH) && !this.cc.canMove(Direction.SOUTH) &&
+        this.cc.canMove(Direction.WEST) && this.cc.canMove(Direction.EAST)) {
+        blocking = true;
+    } else if (!this.cc.canMove(Direction.WEST) && !this.cc.canMove(Direction.EAST) &&
+        this.cc.canMove(Direction.NORTH) && this.cc.canMove(Direction.SOUTH)) {
+        blocking = true;
+    }
+
+    //we have a wall on either side, but the direction to the exit is clear.
+    //WE should be blocking the hallway now, waiting for whoever to come through.
+    if (blocking && this.cc.canMove(toExit)) {
```

Comment:
I was thinking best case it would find a hall that goes straight at the exit, which would probably be close to the exit, worst case it would make it all the way to the exit. If i allowed it to just be blocking any space, it could end up a long way away from where the player is going to go more easily i thought.

raph


On 3/9/16, at 10:37, Bovard Doerschuk-Tiberi <notifications@github.com> wrote:

In lib/CreepPlayers/Behaviors/MoveToExitHallwayDumb.js <https://github.com/bovard/raid/pull/46#discussion_r55565815>:

> +MoveToExitDumb.prototype.act = function() {
> +    var toExit = this.cc.senseDirectionToExit();
> +    var toPlayer = this.cc.getDirectionToPlayer();
> +
> +    var blocking = false;
> +    if (!this.cc.canMove(Direction.NORTH) && !this.cc.canMove(Direction.SOUTH) &&
> +        this.cc.canMove(Direction.WEST) && this.cc.canMove(Direction.EAST)) {
> +        blocking = true;
> +    } else if (!this.cc.canMove(Direction.WEST) && !this.cc.canMove(Direction.EAST) &&
> +        this.cc.canMove(Direction.NORTH) && this.cc.canMove(Direction.SOUTH)) {
> +        blocking = true;
> +    }
> +
> +    //we have a wall on either side, but the direction to the exit is clear.
> +    //WE should be blocking the hallway now, waiting for whoever to come through.
> +    if (blocking && this.cc.canMove(toExit)) {
why blocking && this.cc.canMove(toExit)? what if there is a wall in that direction or soemthing?

E
WWW
 Y
WWW
You are blocking but can't move to exit

—
Reply to this email directly or view it on GitHub <https://github.com/bovard/raid/pull/46/files#r55565815>.